ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/BatchedValidatingCommitter.scala

```diff
@@ -18,6 +18,37 @@ import com.daml.ledger.validator.batch.{
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
+/**
+  * Orchestrates committing to a ledger after validating submissions. Supports parallel validation.
+  * Example usage, assuming a [[com.daml.ledger.participant.state.kvutils.api.BatchingLedgerWriter]] sends the
+  * batched submissions over the wire:
+  * {{{
+  *   ...
+  *   private val ledgerStateOperations = ...
+  *   private val validator = BatchedSubmissionValidator.create(ledgerStateOperations)
+  *   private val validatingCommitter = BatchedValidatingCommitter(
+  *       () => Instant.now(),
+  *       validator)
+  *   ...
+  *
+  *   def commitRequestHandler(request: CommitRequest): Future[CommitResponse] =
+  *     validatingCommitter.commit(
+  *         request.correlationId,
+  *         request.envelope,
+  *         request.participantId,
+  *         ledgerStateOperations)
+  *       .map(...)
+  * }}}
+  *
+  * In case caching is enabled (i.e., [[stateValueCache]] is not a [[Cache.none]]) then for each
```

Comment:
```suggestion
  * When caching is enabled (i.e., [[stateValueCache]] is not a [[Cache.none]]), for each
```

Reply:
I'd go for `If` instead of `When` because this is conditional rather than temporal.